app/mailers/devise_mailer.rb

```diff
@@ -1,11 +1,13 @@
 class DeviseMailer < Devise::Mailer
+  default reply_to: ForemInstance.reply_to_email_address
```

Comment:
**Second pair of eyes needed please 👀**

I've been spent tons of time on a test this in the `devise_mailer_spec.rb` but to no avail. I feel like I've reached a point of tunnel vision and I would really appreciate a second pair of eyes on the issue:

For the purposes of explaining how I went about trying to write the tests, I've removed all the distracting code and just focussed on testing the `reply_to` [in spec/mailers/devise_mailer_spec.rb](https://github.com/forem/forem/blob/main/spec/mailers/devise_mailer_spec.rb):

```ruby
require "rails_helper"

RSpec.describe DeviseMailer, type: :mailer do
  let(:user) { create(:user) }

  describe "#reset_password_instructions" do
    let(:reply_to_email_address) { "custom_reply@forem.com" }
    before do
      allow(Settings::SMTP).to receive(:provided_minimum_settings?).and_return(true)
      allow(Settings::SMTP).to receive(:reply_to_email_address).and_return(reply_to_email_address)
    end

    it "renders reply_to" do
      # Settings::SMTP.provided_minimum_settings? in
      # def self.reply_to_email_address
        # Settings::SMTP.provided_minimum_settings? ? Settings::SMTP.reply_to_email_address : email
      # end
      # evaluates to false and hence its returns the `email` value instead of `Settings::SMTP.reply_to_email_address`
      email = described_class.reset_password_instructions(user, "test")
      expect(email["reply_to"].value).to eq(reply_to_email_address)
    end
  end

  # Since the Settings::SMTP mock was not working correctly, I thought to bypass it with a mocking the entire ForemInstance method as I'm already testing the contents of the method elsewhere.
  describe "#reset_password_instructions" do
    let(:reply_to_email_address) { "custom_reply@forem.com" }
    before do
      allow(ForemInstance).to receive(:reply_to_email_address).and_return(reply_to_email_address)
    end

    it "renders reply_to" do
      email = described_class.reset_password_instructions(user, "test")
      expect(email["reply_to"].value).to eq(reply_to_email_address)
    end
  end
end
```

I get the following failure in both cases above:
<img width="942" alt="Screenshot 2022-02-14 at 09 57 29" src="https://user-images.githubusercontent.com/2786819/153825036-78c83e31-750e-4d95-b6ef-56a39ea59974.png">

Reply:
Thanks Mac! Resolved.